src/components/ProteinViewer/utils.ts

```diff
@@ -1,7 +1,61 @@
 import { createSelector } from 'reselect';
 import { toPlural } from 'utils/pages/toPlural';
 import { NOT_MEMBER_DBS } from 'menuConfig';
+import { getTrackColor, EntryColorMode } from 'utils/entry-color';
 
+const dbs4SingleDomain = [
+  'pfam',
+  'smart',
+  'pirsf',
+  'ncbifam',
+  'hamap',
+  'sfld',
+  'cdd',
+  'profile',
+];
+
+const selectRepresentativeDomains = (domains: Record<string, unknown>[]) => {
+  const flatDomains = [];
+  for (const domain of domains) {
+    const { accession, short_name, source_database, integrated } = domain;
+    for (const location of domain.entry_protein_locations as Array<ProtVistaLocation>) {
+      for (const fragment of location.fragments) {
+        const { start, end } = fragment;
+        flatDomains.push({
+          accession,
+          short_name,
+          source_database,
+          integrated,
+          start,
+          end,
+          color: getTrackColor({ source_database }, EntryColorMode.MEMBER_DB),
+          length: end - start,
```

Comment:
I didn't see `length` used for anything else than comparing domains, but if it is, it should be:

```suggestion
          length: end - start + 1,
```